src/utils/calendar.ts

```diff
@@ -0,0 +1,44 @@
+import ical from "ical-generator";
+
+import { VenueEvent } from "types/venues";
+
+import { eventEndTime, eventStartTime } from "utils/event";
+
+import { WithVenueId } from "utils/id";
+
+import { getFullVenueInsideUrl } from "utils/url";
+
+export interface CreateCalendarProps {
+  calendarName: string;
+  events: WithVenueId<VenueEvent>[];
+}
+
+const createCalendar = ({ calendarName, events }: CreateCalendarProps) => {
+  const cal = ical({ name: calendarName });
+
+  events.forEach((event) =>
+    cal.createEvent({
+      start: eventStartTime(event),
+      end: eventEndTime(event),
+      organizer: `${event.host} <undefined>`, // string format: "name <email>". email cannot be blank
+      description: event.description,
+      summary: event.name,
+      url: getFullVenueInsideUrl(event.venueId),
+    })
+  );
+  return cal;
+};
+
+export const downloadCalendar = ({
+  calendarName,
+  events,
+}: CreateCalendarProps) => {
+  const cal = createCalendar({ calendarName: calendarName, events: events });
```

Comment:
My intent here was that we would pass the already created calendar object as the only param here (replace `???` with the correct type for the generated calendar, this will be the same value used for the `???` for `createCalendar`'s return type:

```suggestion
export const downloadCalendar = (calendar: ???): void => {
```

Reply:
Then, where we call this in `src/components/organisms/NavBarSchedule/NavBarSchedule.tsx`, we would call both `createCalendar` and then `downloadCalendar`

Reply:
We'll still need to replace this with the actual correct type